fedbiomed/researcher/federated_workflows/_federated_workflow.py

```diff
@@ -331,95 +343,93 @@ def info(self, info=None) -> Dict[str, Any]:
                 'Values': []
             }
         info['Arguments'].extend([
-                'Tags',
-                'Nodes filter',
-                'Training Data',
-                'Training Arguments',
-                'Experiment folder',
-                'Experiment Path',
-                'Secure Aggregation'
-            ])
+            'Tags',
+            'Nodes filter',
+            'Training Data',
+            'Training Arguments',
+            'Experiment folder',
+            'Experiment Path',
+            'Secure Aggregation'
+        ])
+
         info['Values'].extend(['\n'.join(findall('.{1,60}', str(e))) for e in [
-                           self._tags,
-                           self._nodes_filter,
-                           self._fds,
-                           self._training_args,
-                           self._experimentation_folder,
-                           self.experimentation_path(),
-                           f'- Using: {self._secagg}\n- Active: {self._secagg.active}'
-                       ]])
+            self._tags,
+            self._nodes_filter,
+            self._fds,
+            self._training_args,
+            self._experimentation_folder,
+            self.experimentation_path(),
+            f'- Using: {self._secagg}\n- Active: {self._secagg.active}'
+        ]])
         print(tabulate.tabulate(info, headers='keys'))
         return info
 
     # Setters
     @exp_exceptions
-    def set_tags(self, tags: Union[List[str], str, None]) -> Union[List[str], None]:
-        """Sets tags + verifications on argument type
-
-        Ensures consistency with the tags attribute following this definition:
-        - if training data is None, then the attributes are consistent
-        - if training data is not None and tags is not None, check that the tags from the training data all contain
-            the tags declared in the self._tags attribute
-
-        The state diagram below explains the flow of setters and consistency checks
-        ![state diagram](https://www.plantuml.com/plantuml/png/ZP91ZzCm48Nl_XLpHjf3usGFqIhQ0ul4XGEik22qCiviQz7OnPvK8CH_9rESAgdjqalhsFE-zsRinq3AqpZinRGWXAUV1_HcG4llhI7uBG2UlJBMsErRHUfbMb4B7vn5Fb7RiDpv8oBb4nAVFRlFQZzYIgbQE7Wy6ZS6E799XF41JVyP4Xka85a2oKoafR84l5ytTv_moy12htKB5BzV-cbZHjStezzvDx0aPJSNRFYc0lRWBBYHj1iGt2ju_35YeDctoVbUNFlTNNTnXwMAT09p4te33mzwvup6hWCXrZm6S4aBUeVwEsXdWmc4Ll_YpCJjAjl3Qn-4td1rSIej67kMKtGFv0uS06tVTP4GDrjOL9_JoZHjqbjCBMzABKfvcV5HcO1FtZlFyQFIXDFRMtHGdpjO2EPEwkiMMWgXvVeg6L-ULpMxHLtSpCvh-lMqWKbnMasQk9Cy7JOS0thuDzqLe4e0LHBucbucUfbxAbbEleRLNzvyNRdKYKkTS_b_kqq2Qgw_x3L9Y4Uq_JZi_m80)
-
-        This function has the following behaviour:
+    def set_tags(
+        self,
```

Comment:
@srcansiz thank you for the review ! Here are several points in `set_tags()` I think relevant to discuss and maybe tune

1. I'm not sure if this is a good idea to prevent to set to `None` because it breaks backward compatibility while not really needed. But if you think appropriate, we can keep it. In any case, if we prevent `None` we should update the error strings accordingly (not `... a str, a list of str, or None`).

2. what is the purpose of additional test `if not tags:` ? Most badly typed args will execute:
    ```
        else:
            raise FedbiomedTypeError(
                f"{ErrorNumbers.FB410.value} `tags` must be a str, a list of str, or None")
    ```
    ... except if we have empty list or empty string.
    So if we want to check that case then we probably should have a coherent error message eg "`tags` must be a non-empty str, or a non-empty list of str"

3. naming `reset_training_data` and corresponding docstrings: I would suggest to name it rather `update_training_data` or equivalent. The name "reset" is rather misleading as it suggests returning to some initial value.

Reply:
Hi @mvesin, 

Thank you for your comments.

1 - It seems to me that setting tags to `None` is not a good idea or does not make any sense. It was implemented that way because experiment instantiation was allowing to set tags to `None`. And `set_training_data` (which depends on tags) raises error if the tags are `None`. So why do we allow setting tags to `None`. Anyway, you are right that it is not a good idea to break the API while it is not needed, so I can change it to raise deprecation warning when the tags are set to None.  

2 - As you said `if not tags` makes sure that the list or string is not empty. And the error message is 
```
f"{ErrorNumbers.FB410.value}: Invalid value for tags argument {tags}, tags " 
"should be list of str o str.")
```
and I agree that it is a good idea to add `non-empty list`, `non-empty str` to the message.

3 - I used `reset` term more like set again training data again using new tags. It is true that reset is conflicting with the doc string. It is okay for me to change it as `update_..`.

4 - I removed those because it seems unnecessary to me. Basically we are raising an error to catch it and printed with as nicely (thanks to exp_exception) but also calling logger.critical to show to same output. So that's why i wanted to remove it. If you want me to make it coherent I can revert it to call logger.critical again or remove all logger.critical calls and just raise an error.